data-upgrade-processes-documents/src/main/java/org/exoplatform/migration/ProcessesDocumentsUpgradePlugin.java

```diff
@@ -0,0 +1,259 @@
+  }
+
+  private List<AttachmentContextEntity> getEntityAttached(String uuid) {
+    RequestLifeCycle.begin(this.entityManagerService);
+    EntityManager entityManager = this.entityManagerService.getEntityManager();
+    List<AttachmentContextEntity> results = new ArrayList<>();
+    try {
+      String sqlString = "SELECT * FROM EXO_ATTACHMENTS_CONTEXT WHERE ATTACHMENT_ID = '"+uuid+"'";
+      jakarta.persistence.Query query = entityManager.createNativeQuery(sqlString, AttachmentContextEntity.class);
+      results = query.getResultList();
+    } catch (Exception e) {
+      log.error("Error when reading attachment entity for file with id={}", uuid,e);
+    } finally {
+      RequestLifeCycle.end();
+    }
+    return results;
+  }
+
+  private List<AttachmentContextEntity> getAllAttachments() {
+    RequestLifeCycle.begin(this.entityManagerService);
+    EntityManager entityManager = this.entityManagerService.getEntityManager();
+    List<AttachmentContextEntity> results = new ArrayList<>();
+    try {
+      String sqlString = "SELECT * FROM EXO_ATTACHMENTS_CONTEXT";
+      jakarta.persistence.Query query = entityManager.createNativeQuery(sqlString, AttachmentContextEntity.class);
+      results = query.getResultList();
+    } catch (Exception e) {
+      log.error("Error when reading all attachments entity",e);
+    } finally {
+      RequestLifeCycle.end();
+    }
+    return results;
+  }
+
+private void updateAttachementUuId(String oldUUID, String newUUID) {
+    RequestLifeCycle.begin(this.entityManagerService);
+    EntityManager entityManager = this.entityManagerService.getEntityManager();
+    try {
+      entityManager.getTransaction().begin();
+      String sqlString = "UPDATE EXO_ATTACHMENTS_CONTEXT SET ATTACHMENT_ID = '"+newUUID+"' WHERE ATTACHMENT_ID='"+oldUUID+"'";
+      jakarta.persistence.Query query = entityManager.createNativeQuery(sqlString, AttachmentContextEntity.class);
+      query.executeUpdate();
+      entityManager.getTransaction().commit();
+    } catch (Exception e) {
+      log.error("Error when updating attachment for attachmentContext with uuid={}", oldUUID,e);
+    } finally {
+      RequestLifeCycle.end();
+    }
+  }
+
+
+  private Node duplicateItem(Node oldNode, Node destinationNode, Node parentNode) throws Exception{
+    Node newNode;
+    String name = oldNode.getName();
+    String title = oldNode.getProperty("exo:title").getString();
+    if (((NodeImpl) destinationNode).getIdentifier().equals(((NodeImpl) parentNode).getIdentifier())){
+      name = name.replace(".docxf", ".pdf");
+      name = name.replace(".oform", ".pdf");
+      title = title.replace(".docxf", ".pdf");
+      title = title.replace(".oform", ".pdf");
+      String newName = name;
+      int i =0;
+      while((destinationNode.hasNode(newName))){
+        i++;
+        newName = name + " (" + i + ")";
+      }
+      name = newName.toLowerCase();
+      if(i>0){
+        title = title + " (" + i + ")";
+      }
```

Comment:
Won't it be better to put the extension after the index of the file ?

Reply:
As this case is very minoritary, and as it requires further tests, I prefer to let like it